cmd/zdb/zdb.c

```diff
@@ -3606,8 +3606,7 @@ dump_l2arc_log_blocks(int fd, l2arc_dev_hdr_phys_t l2dhdr)
 		}
 
 		if (this_lb.lb_magic == BSWAP_64(L2ARC_LOG_BLK_MAGIC))
-			byteswap_uint64_array(&this_lb, psize);
-
+			byteswap_uint64_array(&this_lb, sizeof (this_lb));
```

Comment:
Good catch; I had to go look at the actual implementation of byteswap_uint64_array() to check that this was correct.  The 'size' parameter name is dangerously ambiguous and personally I'd like it to be called 'num_bytes' or something - out of scope of this change.